src/preferences/preferences.cpp

```diff
@@ -1550,6 +1550,37 @@ const std::vector<game_config::server_info>& prefs::builtin_servers_list()
 	return pref_servers;
 }
 
+void prefs::add_game_preset(config& preset)
```

Comment:
Should be an rvalue reference if this function is meant to take ownership.

Reply:
It can't be const though - the id key is added.

Reply:
In that case, I'd suggest creating the `game_preset` child in this function instead and simply taking as input its contents. That would save you from having to validate the input.
```cpp
void preferences::add_game_preset(config&& preset_data)
{
	config preset{ prefs_list::game_preset, std::move(preset_data) };

	int min = 0;
	for(const auto& c : preferences_.child_range(prefs_list::game_preset)) {
		min = std::min(min, c["id"].to_int());
	}

	preset["id"] = min - 1;
	preferences_.append(std::move(preset));
}
```

Then in `mp_create_game::save_preset()`, you just create `root` with the attributes, no `add_child`.

Reply:
Updated.